core/src/test/java/org/apache/calcite/util/SourceTest.java

```diff
@@ -158,16 +157,16 @@ void testAbsoluteFileToUrl(String path, String expectedUrl) throws URISyntaxExce
     // If the file is located in a JAR, we cannot write the file in place
     // so we add it to the /tmp directory
     // the expected output is /tmp/[jarname]/[path-to-file-in-jar/filename]_actual.json
-    logFilePath = refFilePath.replaceAll(
+    logFilePath = refFilePath.replace(
             ".*\\/(.*)\\.jar\\!(.*)\\.json",
             "/tmp/$1$2_actual.json");
     final File logFile = new File(logFilePath);
     assertThat(refFile, not(is(logFile.getAbsolutePath())));
     boolean b = logFile.getParentFile().mkdirs();
     Util.discard(b);
-    try (FileWriter fw = new FileWriter(logFile);
-         PrintWriter pw = new PrintWriter(fw)) {
-      pw.println("hello, world!");
+    try  {
+      FileOutputStream fw = new FileOutputStream(logFilePath);
+      fw.write("hello, world!\n".getBytes(StandardCharsets.UTF_8));
```

Comment:
Changes here to keep `forbiddenApis` happy. There is a [suppress annotation](https://github.com/policeman-tools/forbidden-apis/issues/56) we could try but I think we'd need to add some `org.elasticsearch.common` as a dependency and this seemed easier. It does not appear to meaningfully alter the test.